### PR TITLE
MAINT: Fix conditionals for PRs

### PR DESCRIPTION
**Pull request by larsoner**
Let's see if dipy 3.10 is likely to fix our problems or if there is something else we can know in advance we need to investigate

### CODE REVIEW

**Comment by larsoner** [comment]
... I'll push a commit to take care of it

**Comment by hoechenberger** [comment]
@larsoner You'd have to patch `construct.yml` as well if you don't fetch the certs … as they are mentioned in this config file
I don't think it's worth adding this level of complexity!

**Comment by hoechenberger** [comment]
@larsoner Let's keep the signing but disable the notarization in PRs, WDYT?

**Comment by larsoner** [comment]
I am not entirely clear on the differences here, but I would not allow PRs to use any steps that require secrets. If you can sign without secrets and need secrets to notarize, then I agree that seems reasonable

**Comment by hoechenberger** [comment]
Like I said, the only way to skip signing is by patching `construct.yaml` as this file specifies IDs of signing keys. If the keys cannot be found, the build fails.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -100,13 +100,19 @@ jobs:
         run: |
           patch $CONDA_PREFIX/lib/python3.9/site-packages/constructor/osxpkg.py ./assets/constructor_macOS.patch
 
+      - name: Patch config (macOS pull request)
+        if: ${{ runner.os == 'macOS' && github.event_name == 'pull_request' }}
+        shell: bash -l {0}
+        run: |
+          sed -i "s/9779L28NP8//" recipes/mne-python_1.0/construct.yaml
```
Comment (hoechenberger) [inline .github/workflows/build.yml:107]: macOS ;))

```suggestion
          sed -i "" "s/9779L28NP8//" recipes/mne-python_1.0/construct.yaml
```

**Comment by hoechenberger** [comment]
@larsoner This is nicer though: https://github.com/hoechenberger/mne-installers/pull/78#discussion_r826290194

**Review by hoechenberger — APPROVED** [review]

**Pull request merged**